File: Lib/Models/Job.cs
namespace Lib.Models;

public class Job : BaseModel
{
    public Guid JobId { get; set; }
    public Guid CategoryId { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public string Data { get; set; } //json blob
}

public static class JobExtensions
{
    public static Job UpdateWith(this Job existing, Job updated)
    {
        existing.Name = updated.Name;
        existing.Description = updated.Description;
        existing.Data = updated.Data;
        return existing;
    }
}

File: App/ClientApp/src/app/core/models/job.ts
export interface Job {
  jobId: string;
  categoryId: string;
  name: string;
  description: string;
  data: string;
  // BaseModel properties
  createdAt: Date;
  modifiedAt: Date;
  userId: string;
  concurrencyStamp: string;
}

export const defaultJob: Job = {
  jobId: "",
  categoryId: "",
  name: "",
  description: "",
  data: "",
  createdAt: new Date(),
  modifiedAt: new Date(),
  userId: "",
  concurrencyStamp: "",
};
